eyelid-match-ops/src/primitives/yashe/conf.rs

```diff
@@ -25,27 +29,67 @@ use crate::primitives::poly::modular_poly::conf::TinyTest;
 /// Encryption keys and ciphertexts with different parameters are incompatible.
 pub trait YasheConf: PolyConf
```

Comment:
An alternative: instead of YasheConf being a super-trait of PolyConf, it defines two PolyConf as associated types, for plaintext and ciphertexts.

```rust
pub trait YasheConf {
    type PlainPolyConf: PolyConf;

    type CipherPolyConf: PolyConf;

    …
}

```

Reply:
PolyConf has 2 items: degree and modulus. But degree must be the same for PlainPolyConf and CipherPolyConf. YasheConf must have a way to configure this common degree, and a way to choose the plaintext and the ciphertext modulus (denoted t and q in the paper). Beyond that, YasheConf has other parameters like standard deviations. It can grow in the future if we start doing things like bootstrapping.

Reply:
Let’s open a ticket for this?
I don’t mind which way we do it, but some ways will be easier in Rust than others.

Reply:
[Here](https://github.com/Inversed-Tech/eyelid/issues/90) it is.